auraed/src/cells/cell_service/cells/cell_graph.rs

```diff
@@ -1,24 +1,46 @@
 use super::{CellName, CellSpec};
 
-#[derive(Clone)]
-pub struct GraphNode {
+pub struct Valid {
     pub cell_info: Option<(CellName, CellSpec)>,
     pub children: Vec<GraphNode>,
 }
 
+#[derive(Clone)]
+pub enum GraphNode {
+    Valid { cell_info: Option<(CellName, CellSpec)>, children: Vec<GraphNode> },
+    Invalid { cell_name: CellName },
+}
+
 impl GraphNode {
     pub fn with_cell_info(
         self,
         cell_name: &CellName,
         cell_spec: &CellSpec,
     ) -> GraphNode {
-        GraphNode {
-            cell_info: Some((cell_name.clone(), cell_spec.clone())),
-            children: self.children,
+        match self {
+            GraphNode::Valid { cell_info: _, children } => GraphNode::Valid {
+                cell_info: Some((cell_name.clone(), cell_spec.clone())),
+                children: children,
+            },
+            GraphNode::Invalid { cell_name: _ } => self,
         }
     }
 
-    pub fn with_children(self, children: Vec<GraphNode>) -> GraphNode {
-        GraphNode { cell_info: self.cell_info.as_ref().cloned(), children }
+    pub fn with_children(self, new_children: Vec<GraphNode>) -> GraphNode {
+        match self {
+            GraphNode::Valid { cell_info, children: _ } => GraphNode::Valid {
+                cell_info: cell_info.as_ref().cloned(),
+                children: new_children,
+            },
+            GraphNode::Invalid { cell_name: _ } => self,
+        }
+    }
+
+    pub fn unwrap_valid(self) -> Valid {
+        if let GraphNode::Valid { cell_info, children } = self {
+            Valid { cell_info, children }
+        } else {
+            panic!("invalid graphnode")
```

Comment:
This and the other panics will need to be eliminated. AKA, if we can't handle the error, the error should flow back up to the client.